jbmc/src/java_bytecode/java_entry_point.cpp

```diff
@@ -447,77 +460,95 @@ exprt::operandst java_build_arguments(
   return main_arguments;
 }
 
-void java_record_outputs(
+/// Mark return value, pointer type parameters and the exception as outputs.
+static code_blockt java_record_outputs(
   const symbolt &function,
   const exprt::operandst &main_arguments,
-  code_blockt &init_code,
   symbol_table_baset &symbol_table)
 {
-  const java_method_typet::parameterst &parameters =
-    to_java_method_type(function.type).parameters();
-
-  bool has_return_value =
-    to_java_method_type(function.type).return_type() != java_void_type();
+  code_blockt init_code;
 
-  if(has_return_value)
+  if(auto return_value = record_return_value(function, symbol_table))
   {
-    // record return value
-    codet output(ID_output);
-    output.operands().resize(2);
+    init_code.add(std::move(*return_value));
+  }
 
-    const symbolt &return_symbol=
-      *symbol_table.lookup(JAVA_ENTRY_POINT_RETURN_SYMBOL);
+  init_code.append(
+    record_pointer_parameters(function, main_arguments, symbol_table));
 
-    output.op0()=
-      address_of_exprt(
-        index_exprt(
-          string_constantt(return_symbol.base_name),
-          from_integer(0, index_type())));
-    output.op1()=return_symbol.symbol_expr();
-    output.add_source_location()=function.location;
+  init_code.add(record_exception(function, symbol_table));
 
-    init_code.add(std::move(output));
-  }
+  return init_code;
+}
 
-  for(std::size_t param_number=0;
-      param_number<parameters.size();
+static optionalt<codet> record_return_value(
+  const symbolt &function,
+  const symbol_table_baset &symbol_table)
+{
+  if(to_java_method_type(function.type).return_type() == java_void_type())
+    return nullopt;
```

Comment:
:pick: Why not `return {}`?

Reply:
:man_shrugging: I have switched it back to `return {}`. At the time I wrote this I was thinking that perhaps `nullopt` was used more in CBMC. However I have now actually checked for the number of occurrences of both and `{}` is used far more widely. So using `{}` is more consistent with the rest of the code base. The preference for one or the other feels like the sort of thing which should probably be in the coding standards. However the coding standards don't mention `optionalt`.